### PR TITLE
[BUG] Prevent server freeze when FT.SEARCH timeout while sorting

### PR DESCRIPTION
**Pull request by ashtul**

### CODE REVIEW

File: src/aggregate/aggregate_exec.c [hunk 1]
```diff
@@ -190,8 +190,7 @@ void sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
 
   rc = rp->Next(rp, &r);
   long resultsLen = REDISMODULE_POSTPONED_ARRAY_LEN;
-  if (rc == RS_RESULT_TIMEDOUT && !(req->reqflags & QEXEC_F_IS_CURSOR) && !IsProfile(req) &&
-      RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Fail) {
+  if (rc == RS_RESULT_TIMEDOUT && !(req->reqflags & QEXEC_F_IS_CURSOR) && !IsProfile(req)) {
```
Comment (MeirShpilraien) [inline src/aggregate/aggregate_exec.c:193]: Can you explain this change, only if `RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Fail` you want to return a single results which is the timeout no?

File: src/aggregate/aggregate_exec.c [hunk 2]
```diff
@@ -211,7 +210,6 @@ void sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
         RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Fail) {
       RedisModule_ReplyWithSimpleString(outctx, "Timeout limit was reached");
     } else {
-      rc = RS_RESULT_OK;
```
Comment (MeirShpilraien) [inline src/aggregate/aggregate_exec.c:214]: Why is this not considered ok?

**Review by MeirShpilraien — COMMENTED** [review]
Can you better explain the fix? What was the issue and on what cases it occurred?

File: src/result_processor.c [hunk 3]
```diff
@@ -352,7 +352,8 @@ static int rpsortNext_innerLoop(ResultProcessor *rp, SearchResult *r) {
   }
 
   // if our upstream has finished - just change the state to not accumulating, and yield
-  if (rc == RS_RESULT_EOF) {
+  if (rc == RS_RESULT_EOF ||
+      (rc == RS_RESULT_TIMEDOUT && RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Return)) {
```
Comment (ashtul) [inline src/result_processor.c:356]: @MeirShpilraien 
The logic change and in the case of `TimeoutPolicy_Return` we do not return `RS_RESULT_TIMEDOUT` but `RS_RESULT_OK`.
This change answers your question from above.

Comment (MeirShpilraien) [reply src/result_processor.c:356]: But you assume you always have a sorter that will catch the timeout no? This is not the case with FT.AGGREGATE and I believe its a shared code with FT.AGGREGATE no?

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**